homeassistant/components/light/niko_home_control.py

```diff
@@ -0,0 +1,99 @@
+"""
+Support for Niko Home Control.
+
+For more details about this platform, please refer to the documentation
+https://home-assistant.io/components/light.niko_home_control/
+"""
+import logging
+
+import voluptuous as vol
+
+# Import the device class from the component that you want to support
+from homeassistant.components.light import (
+    ATTR_BRIGHTNESS, Light, PLATFORM_SCHEMA)
+from homeassistant.const import CONF_HOST
+import homeassistant.helpers.config_validation as cv
+
+# Home Assistant depends on 3rd party packages for API specific code.
+REQUIREMENTS = ['niko-home-control==0.1.8']
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_HOST): cv.string,
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup the NikoHomeControl platform."""
+    import nikohomecontrol
+
+    host = config.get(CONF_HOST)
+
+    hub = nikohomecontrol.Hub({
```

Comment:
I assume that there will be an exception if the hub is not available or the IP address wrong. You should catch it and log something.

You could also use `PlatformNotReady` if it make sense.